examples/kademlia-example.rs

```diff
@@ -0,0 +1,157 @@
+#![allow(dead_code, unused_variables)]
```

Comment:
Instead of an example, could you add a test to `kad/src/behaviour/tests.rs`?

Reply:
Writing this test in the `kademlia` module will be kind of difficult, as I'm using a lot of other things as well, for building the swarm or transport. Is it okay if I put this is in a test directory?

Reply:
> Writing this test in the kademlia module will be kind of difficult, as I'm using a lot of other things as well, for building the swarm or transport. Is it okay if I put this is in a test directory?

No need for most of it. You can take the `manual_bucket_insert` and `bootstrap` test as an example.

> I can further enhance kademlia-example to create four peers and cross-check if the peer in client mode is actually, absent from the other's routing tables.

I don't think there is any need for an example. The "cross-check" sounds great, but should really be done in a test instead of an example.